src/headless/main.cpp

```diff
@@ -52,37 +57,92 @@ static bool headless_load(Dataloader::path_vector_t const& roots) {
 		ret = false;
 	}
 
+	return ret;
+}
+
+static bool run_headless(Dataloader::path_vector_t const& roots, bool run_tests) {
+	bool ret = true;
+
+	setup_logger_funcs();
+
+	Dataloader dataloader;
+	if (!dataloader.set_roots(roots)) {
+		Logger::error("Failed to set dataloader roots!");
+		ret = false;
+	}
+
+	GameManager game_manager { []() {
+		Logger::info("State updated");
+	} };
 
-	Testing testing = Testing(&game_manager);
-	std::cout << std::endl << "Testing Loaded" << std::endl << std::endl;
-	testing.execute_all_scripts();
-	testing.report_results();
-	std::cout << "Testing Executed" << std::endl << std::endl;
+	ret &= headless_load(game_manager, dataloader);
+
+	if (run_tests) {
+		Testing testing = Testing(&game_manager);
+		std::cout << std::endl << "Testing Loaded" << std::endl << std::endl;
+		testing.execute_all_scripts();
+		testing.report_results();
+		std::cout << "Testing Executed" << std::endl << std::endl;
+	}
 
 	return ret;
 }
 
+/*
+	$ program [-h] [-t] [-b] [path]+
+*/
+
 int main(int argc, char const* argv[]) {
-	Dataloader::path_vector_t roots;
-	if (argc < 2) {
-		// TODO - non-Windows default paths
-		static const fs::path default_path = "C:/Program Files (x86)/Steam/steamapps/common/Victoria 2";
-
-		std::cout
-			<< "Usage: " << get_program_name(argc > 0 ? argv[0] : nullptr) << " <base defines dir> [[mod defines dir]+]\n"
-			<< "Requires defines path(s) as arguments, starting with the base defines and continuing with mods.\n"
-			<< "(Paths with spaces need to be enclosed in \"quotes\").\n"
-			<< "Defaulting to " << default_path << std::endl;
-		roots.push_back(default_path);
-	} else {
-		for (int i = 1; i < argc; ++i) {
-			roots.push_back(argv[i]);
+	char const* program_name = get_program_name(argc > 0 ? argv[0] : nullptr);
+
+	fs::path root;
+	bool run_tests = false;
+
+	int argn = 0;
+	while (++argn < argc) {
+		char const* arg = argv[argn];
+		if (strcmp(arg, "-h") == 0) {
+			print_help(program_name);
+			return 0;
+		} else if (strcmp(arg, "-t") == 0) {
+			run_tests = true;
+		} else if (strcmp(arg, "-b") == 0) {
+			if (root.empty())  {
+				if (++argn < argc) {
+					root = argv[argn];
+					if (!root.empty()) {
+						continue;
+					} else {
+						std::cerr << "Empty path after base directory command line argument \"-b\"." << std::endl;
+					}
+				} else {
+					std::cerr << "Missing path after base directory command line argument \"-b\"." << std::endl;
+				}
+			} else {
+				std::cerr << "Duplicate base directory command line argument \"-b\"." << std::endl;
+			}
+			print_help(program_name);
+			return -1;
+		} else {
+			break;
+		}
+	}
+	if (root.empty()) {
+		root = Dataloader::search_for_game_path();
```

Comment:
Probably should allow sending a path to this as a command line argument, `search_for_game_path` is basically already designed to allow user input to almost anywhere inside that'll point to a valid path of something Steam related that can confirm Victoria 2 is installed and open its path, and it ensures that if you point to a Victoria 2 path, it immediately returns it. (and if the path fails anyway, it tries an empty hint_path)

Reply:
I'll add another argument that passes a path to `Dataloader::search_for_game_path`, I was gonna make `-b` do it but that wouldn't work for our custom assets which won't have the exe in the folder.

Reply:
Yeah the entire point of `search_for_game_path` is to find the game itself, its a lot easier to do things according to that hence why its built that way.